File: Examples/iOS/AudioKitDemo/AudioKitDemo/Synthesis/SynthesisViewController.m
//
//  SynthesisViewController.m
//  AudioKitDemo
//
//  Created by Aurelius Prochazka on 2/14/15.
//  Copyright (c) 2015 Aurelius Prochazka. All rights reserved.
//

#import "SynthesisViewController.h"
#import "AKFoundation.h"
#import "FMSynthesizer.h"


@implementation SynthesisViewController
{
    IBOutlet UIView *fmSynthesizerTouchView;
    IBOutlet UIView *tambourineTouchView;
    
    AKTambourineInstrument *tambourine;
    FMSynthesizer *fmSynthesizer;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    tambourine = [[AKTambourineInstrument alloc] init];
    [AKOrchestra addInstrument:tambourine];
    
    AKAmplifier *amp = [[AKAmplifier alloc] initWithInput:tambourine.output];
    [AKOrchestra addInstrument:amp];
    [amp start];

    fmSynthesizer = [[FMSynthesizer alloc] init];
    [AKOrchestra addInstrument:fmSynthesizer];
}

- (IBAction)tapTambourine:(UITapGestureRecognizer *)sender {

    CGPoint touchPoint = [sender locationInView:tambourineTouchView];
    float scaledX = touchPoint.x / tambourineTouchView.bounds.size.width;
    float scaledY = 1.0 - touchPoint.y / tambourineTouchView.bounds.size.height;
    
    float intensity = scaledY*4000 + 20;
    float dampingFactor = scaledX / 2.0;
    AKTambourineNote *note = [[AKTambourineNote alloc] initWithIntensity:intensity
                                                           dampingFactor:dampingFactor];
    [tambourine playNote:note];
}


- (IBAction)tapFMOscillator:(UITapGestureRecognizer *)sender {
    
    CGPoint touchPoint = [sender locationInView:fmSynthesizerTouchView];
    float scaledX = touchPoint.x / fmSynthesizerTouchView.bounds.size.width;
    float scaledY = 1.0 - touchPoint.y / fmSynthesizerTouchView.bounds.size.height;
    
    float frequency = scaledY*400;
    float color = scaledX;
    FMSynthesizerNote *note = [[FMSynthesizerNote alloc] initWithFrequency:frequency color:color];
    [fmSynthesizer playNote:note];
}

@end

File: Examples/iOS/Swift/AudioKitDemo/AudioKitDemo/Synthesis/SynthesisViewController.swift
//
//  ViewController.swift
//  AudioKitDemo
//
//  Created by Nicholas Arner on 3/1/15.
//  Copyright (c) 2015 AudioKit. All rights reserved.
//


class SynthesisViewController: UIViewController {
    
    @IBOutlet var fmSynthesizerTouchView: UIView!
    @IBOutlet var tambourineTouchView: UIView!
    
    let tambourine    = AKTambourineInstrument()
    let fmSynthesizer = FMSynthesizer()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the vFMSynthesizeriew, typically from a nib.
        AKOrchestra.addInstrument(tambourine)
        AKOrchestra.addInstrument(fmSynthesizer)
        
        let amp = AKAmplifier(input: tambourine.output)
        AKOrchestra.addInstrument(amp)
        amp.start()
    }


    @IBAction func tapTambourine(sender: UITapGestureRecognizer) {
        
        let touchPoint = sender.locationInView(tambourineTouchView)
        let scaledX = touchPoint.x / tambourineTouchView.bounds.size.height
        let scaledY = 1.0 - touchPoint.y / tambourineTouchView.bounds.size.height
        
        let intensity = Float(scaledY*4000 + 20)
        let dampingFactor = Float(scaledX / 2.0)
        
        let note = AKTambourineNote(intensity: intensity, dampingFactor: dampingFactor)
        tambourine.playNote(note)
    }
    
    @IBAction func tapFMOscillator(sender: UITapGestureRecognizer) {
        
        let touchPoint = sender.locationInView(fmSynthesizerTouchView)
        let scaledX = touchPoint.x / fmSynthesizerTouchView.bounds.size.height
        let scaledY = 1.0 - touchPoint.y / fmSynthesizerTouchView.bounds.size.height
        
        let frequency = Float(scaledY*400)
        let color = Float(scaledX)
        
        let note = FMSynthesizerNote(frequency: frequency, color: color)
        fmSynthesizer.playNote(note)
    }
}
